src/test/java/com/iota/iri/conf/ConfigFactoryTest.java

```diff
@@ -0,0 +1,170 @@
+package com.iota.iri.conf;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Tests for the {@link ConfigFactory}
+ */
+public class ConfigFactoryTest {
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    /**
+     * Creates and validates a Testnet {@link IotaConfig}.
+     */
+    @Test
+    public void createIotaConfigTestnet() {
+        IotaConfig iotaConfig = ConfigFactory.createIotaConfig(true);
+        assertTrue("Expected iotaConfig as instance of TestnetConfig.", iotaConfig instanceof TestnetConfig);
+        assertTrue("Expected iotaConfig as Testnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Mainnet {@link IotaConfig}.
+     */
+    @Test
+    public void createIotaConfigMainnet() {
+        IotaConfig iotaConfig = ConfigFactory.createIotaConfig(false);
+        assertTrue("Expected iotaConfig as instance of MainnetConfig.", iotaConfig instanceof MainnetConfig);
+        assertFalse("Expected iotaConfig as Mainnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Testnet {@link IotaConfig} with <code>TESTNET=true</code> in config file and
+     * <code>testnet: false</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetTrueAndFalse() throws IOException {
+        // lets assume in our configFile is TESTNET=true
+        File configFile = createTestnetConfigFile("true");
+
+        // but the parameter is set to testnet=false
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, false);
+        assertTrue("Expected iotaConfig as instance of TestnetConfig.", iotaConfig instanceof TestnetConfig);
+        assertTrue("Expected iotaConfig as Testnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Testnet {@link IotaConfig} with <code>TESTNET=true</code> in config file and
+     * <code>testnet: true</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetTrueAndTrue() throws IOException {
```

Comment:
Maybe you can delete https://github.com/iotaledger/iri/blob/dev/src/test/java/com/iota/iri/conf/ConfigTest.java#L82
and https://github.com/iotaledger/iri/blob/dev/src/test/java/com/iota/iri/conf/ConfigTest.java#L150

If you want I think using `AssertThat` gives nicer error messages. The way you did it is fine as well.